src/pngchunk_int.cpp

```diff
@@ -656,8 +663,13 @@ namespace Exiv2 {
             }
         }
         sp++ ; // step over '\n'
+        if (sp == eot) {
+            return DataBuf();
+        }
 
         long length = (long) atol(startOfLength);
+        enforce(length >= 0, Exiv2::kerCorruptedMetadata);
```

Comment:
Could we merge these 2 conditions into a single call to `enforce` ? I am fine if you prefer to leave it as it is.

Reply:
done